### PR TITLE
Update CqlLoader logic to cope with strings containing semicolons

### PR DESCRIPTION
**Pull request by SebDuff**
Also added unit tests to cover escaped quotes inside of strings

### CODE REVIEW

File: src/main/java/uk/sky/cqlmigrate/CqlLoader.java [hunk 1]
```diff
@@ -26,9 +31,9 @@ static void load(Session session, Path cqlPath) {
             cqlReader.lines().forEach(statementBuilder::append);
 
             String cqlStatements = statementBuilder.toString();
-            checkState(cqlStatements.endsWith(";"), "had a non-terminated cql line: %s", cqlStatements);
+            checkState(cqlStatements.endsWith(CQL_STATEMENT_TERMINATOR), "had a non-terminated cql line: %s", cqlStatements);
```
Comment (chbatey) [inline src/main/java/uk/sky/cqlmigrate/CqlLoader.java]: This looks to throw an exception just be caught below as we're modifying this file we should clean this up

File: src/main/java/uk/sky/cqlmigrate/CqlLoader.java [hunk 2]
```diff
@@ -37,4 +42,21 @@ static void load(Session session, Path cqlPath) {
...
+            if (CharMatcher.is('\'').countIn(candidateStatement) % 2 == 0) {
+                statementList.add(candidateStatement);
+                candidateStatement = "";
+            } else {
+                candidateStatement += CQL_STATEMENT_TERMINATOR;
+            }
+        }
+        return statementList;
+    }
 }
```
Comment (chbatey) [inline src/main/java/uk/sky/cqlmigrate/CqlLoader.java]: fix me :)

**Comment by chbatey** [comment]
We've had to test string parsing logic by side effects to a database due to the coupling of working out what we want to do with the side effect of executing it in a database. 

Can we separate out the gathering the cql statements to execute and executing them so that all the tests with semi colons in strings etc don't need to involve a database?

**Comment by oliverlockwood** [comment]
Good call, I'm working on splitting into two classes `CqlLoader` and `CqlFileParser`.  Stand by.

**Comment by oliverlockwood** [comment]
@chbatey comments addressed, and I've split the logic out.  Two of the tests still make sense in the overall DB-executing `CqlMigratorImplTest`, but one of them has been split into a test for the new isolated class.  See what you think.

**Pull request merged**